samples/Standalone.MvcSample/Startup.cs

```diff
@@ -39,15 +39,33 @@ public void ConfigureServices(IServiceCollection services)
                     }
                     else if (Configuration.GetValue("ActiveLogin:BankId:UseTestEnvironment", false))
                     {
-                        builder.UseTestEnvironment()
-                               .UseClientCertificate(() => new X509Certificate2(Path.Combine(_environment.ContentRootPath, Configuration.GetValue<string>("ActiveLogin:BankId:ClientCertificate"))))
-                               .UseRootCaCertificate(Path.Combine(_environment.ContentRootPath, Configuration.GetValue<string>("ActiveLogin:BankId:CaCertificate:FilePath")));
+                        if(Configuration.GetValue("ActiveLogin:BankId:UseManagedIdentity", false))
+                        {
+                            builder.UseTestEnvironment()
+                                .UseClientCertificateFromManagedIdentityAzureKeyVaultOptions(Configuration.GetSection("ActiveLogin:BankId:ManagedIdentity"))
```

Comment:
Thought a bit, I think you should be specific, so let's have a `ClientCertificate:UseManagedIdentity`.

Reply:
`
public class ClientCertificateFromAzureKeyVaultOptions
{
    public string AzureAdClientId { get; set; }
    public string AzureAdClientSecret { get; set; }
    public string AzureKeyVaultSecretIdentifier { get; set; }
    public bool UseManagedIdentity { get; set; }
}
`

The settings are in conflict if we merge them. The first two strings are not needed for ManagedIdentity, only the key vault uri and identifier are needed here. 

What about a design like this? It makes it easier to understand what settings are required for the different scenarios, and in our docs and samples we only show ManagedIdentity version to promote that usage. The credentials version is only shown here for visibility.
I still don't like the "identifier" part because we only support storing the certificate as a secret, not as a certificate.
  
`
builder.UseTestEnvironment()
    .UseClientCertificateFromAzureKeyVault(certificateOptions => certificateOptions.UseManagedIdentity(keyVaultUri: "", certificateIdentifier: ""))
    .UseClientCertificateFromAzureKeyVault(certificateOptions => certificateOptions.UseCredentials(azureAdClientId: "", azureAdClientSecret: "", certificateIdentifier: ""));
`

Reply:
> The settings are in conflict if we merge them. The first two strings are not needed for ManagedIdentity, only the key vault uri and identifier are needed here.

Yes, I agree, it wouldn't bee super clear, but `UseManagedIdentity` would "control" what parameters is needed.

> What about a design like this? It makes it easier to understand what settings are required for the different scenarios, and in our docs and samples we only show ManagedIdentity version to promote that usage. The credentials version is only shown here for visibility.

I like the pattern in general, and we use it for the `.AddBankId()` things, but in this case I think's it would be really nice just to bind to your config as we have now. As stated, `keyVaultUri` and `certificateIdentifier` combined are exactly the same as `certificateIdentifier`. We should choose one of those paths, where `certificateIdentifier` might be the easiest. Is `UseManagedIdentity` good enough?



> I still don't like the "identifier" part because we only support storing the certificate as a secret, not as a certificate.

Technically, it is stored as a certificate in Azure KeyVault, but for reasons I can't remember in detail, we had to retrieve it as Secret. I think that if we received it as certificate it was altered in any way, but using the secret endpoint we retrieve the raw data.

![image](https://user-images.githubusercontent.com/1662918/53053922-13528700-34a3-11e9-854b-8b82ef1c1fec.png)

Reply:
@PeterOrneholm is right, it stored in the same storage, but the only difference is for Certificates you need to change a bit the identifier, so our existing solution works already for both Secrets and Certificates